haystack/dataclasses/breakpoints.py

```diff
@@ -6,91 +6,65 @@
 from typing import Optional, Union
 
 
-@dataclass
+@dataclass(frozen=True)
```

Comment:
I opted to add `frozen=True` since this will automatically trigger dataclass to add `hash` and `eq` methods so we don't need to make our own. Also I don't see the need to mutate a breakpoint once it's made.